arangod/IResearch/IResearchViewSort.h

```diff
@@ -89,32 +84,42 @@ class IResearchViewSort {
     return true;
   }
 
-  std::vector<std::vector<basics::AttributeName>> const& fields()
-      const noexcept {
-    return _fields;
-  }
+  auto const& fields() const noexcept { return _fields; }
 
   std::vector<basics::AttributeName> const& field(size_t i) const noexcept {
     TRI_ASSERT(i < this->size());
-
     return _fields[i];
   }
 
   bool direction(size_t i) const noexcept {
     TRI_ASSERT(i < this->size());
-
     return _directions[i];
   }
 
-  size_t memory() const noexcept;
-
   bool toVelocyPack(velocypack::Builder& builder) const;
   bool fromVelocyPack(velocypack::Slice, std::string& error);
 
+ protected:
+  void clear() noexcept {
```

Comment:
should "clear" and "memory" be virtual? To be on safe side if we will work through base class pointers?